sheet.go

```diff
@@ -30,11 +30,21 @@ import (
 	"github.com/mohae/deepcopy"
 )
 
+type Option func(ws *xlsxWorksheet)
+
+func WithMaxSheetRow(count int) Option {
+	return func(ws *xlsxWorksheet) {
+		if count > 0 {
+			ws.SheetData.Row = make([]xlsxRow, 0, count)
+		}
+	}
+}
+
 // NewSheet provides the function to create a new sheet by given a worksheet
 // name and returns the index of the sheets in the workbook after it appended.
 // Note that when creating a new workbook, the default worksheet named
 // `Sheet1` will be created.
-func (f *File) NewSheet(sheet string) (int, error) {
+func (f *File) NewSheet(sheet string, options ...Option) (int, error) {
```

Comment:
This function allows users to set cell values ​​randomly when they can't sure the total number of rows. If user know how many rows will be written and data is in ordered, they can using stream writer for get much lower cost. So I doesn't suggest changing the function signature.

Reply:
I saw that using StreamWriter did not help the person who reported issue 876. Maybe I should use both options under the same conditions and compare the results?